### PR TITLE
Support plain number-typed properties in geckolib

### PR DESCRIPTION
**Pull request by heycam**
r? @bholley

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10957)

### CODE REVIEW

File: components/style/values.rs [hunk 1]
```diff
@@ -1477,6 +1477,43 @@ pub mod specified {
...
+
+    impl Number {
+        pub fn parse(input: &mut Parser) -> Result<Number, ()> {
+            parse_number(input).map(Number)
+        }
+
+        fn parse_with_minimum(input: &mut Parser, min: CSSFloat) -> Result<Number, ()> {
+            match parse_number(input) {
+                Ok(value) if value < min => Err(()),
+                value => value.map(Number),
```
Comment (bholley) [inline components/style/values.rs]: Using a newtype as a function in this way seems a bit weird, but maybe it's a totally acceptable pattern. Can you verify that this is acceptable Rust? (No need to block landing on it)

**Comment by bholley** [comment]
Nice!

@bors-servo r=bholley

Don't forget to update the spreadsheet!

**Comment by bholley** [comment]
Oh wait, tidy problem. @bor-servo r-

**Comment by bholley** [comment]
@bors-servo delegate+

I'd recommend running ./mach test-tidy before pushing PRs.

**Comment by ConnorGBrewster** [comment]
@bholley looks like you misspelled bors-servo so the r- didn't go through :/

File: components/style/values.rs (hunk 1 above)
Comment (heycam) [inline components/style/values.rs]: There are a couple of other similar uses in this file – mapping to Au and Percentage – and a bunch more mapping to enum constructors, which is not so different.

**Pull request merged**